dianna/methods/kernelshap_image.py

```diff
@@ -123,12 +127,27 @@ def explain(
             self._runner, np.zeros((len(self.labels), n_segments)))
 
         # Temporarily hide warnings, because shap is very spammy
+        # `shap_values` is a matrix of shapley values for each model outputs
         with LoggingContext(level=logging.CRITICAL):
             shap_values = explainer.shap_values(np.ones(
                 (len(self.labels), n_segments)),
                                                 nsamples=nsamples)
 
-        return shap_values, self.image_segments
+        # create heat_maps where shape is (n_classes, *image_segments.shape)
+        n_classes = len(shap_values)
+        heat_maps = np.zeros((n_classes, *self.image_segments.shape))
+
+        # fill the heat_maps with shap values for each class and segment
+        for i, shap_value in enumerate(shap_values):
+            class_heat_map = heat_maps[i]
+            for index in self.image_segments.flat:
+                class_heat_map[self.image_segments == index] = shap_value[0][index - 1]
```

Comment:
I'm a bit confused about what is in shap_values. It seems there is a list in there for every class. Then inside that list there is another level (hence '[0]') and only in there is the actual list of shap_values for that class. Except for that extra level, it makes sense to me, but maybe we can help the reader of the code a bit with more descriptive variable name?
Maybe shap_value_lists instead of shap_values to imply that there is a multilevel thing in there for instance?

Reply:
Thanks, I agree. It is renamed now.